R/tm_g_ae_oview.R

```diff
@@ -281,83 +285,62 @@ srv_g_ae_oview <- function(id,
           sep = "\n"
         )
       ))
-
-      ANL_UNFILTERED <- datasets$get_data(dataname, filtered = FALSE) # nolint
-      ADSL <- datasets$get_data("ADSL", filtered = TRUE) # nolint
-      ANL <- datasets$get_data(dataname, filtered = TRUE) # nolint
-
-      anl_name <- dataname
-      assign(anl_name, ANL)
-
-      teal.code::chunks_reset(envir = environment())
-
+      ANL <- data[[dataname]]() # nolint
       validate(need(nlevels(ANL[[input$arm_var]]) > 1, "Arm needs to have at least 2 levels"))
       validate_has_data(ANL, min_nrow = 10)
-      if (all(c(input$arm_trt, input$arm_ref) %in% ANL_UNFILTERED[[input$arm_var]])) {
-        validate(
-          need(
-            input$arm_ref %in% ANL[[input$arm_var]],
-            paste0("Selected Control ", input$arm_var, ", ", input$arm_ref, ", is not in the data (filtered out?)")
-          ),
-          need(
-            input$arm_trt %in% ANL[[input$arm_var]],
-            paste0("Selected Treatment ", input$arm_var, ", ", input$arm_trt, ", is not in the data (filtered out?)")
-          )
+      validate(
+        need(
+          input$arm_ref %in% ANL[[input$arm_var]],
+          paste0("Selected Control ", input$arm_var, ", ", input$arm_ref, ", is not in the data (filtered out?)")
+        ),
+        need(
+          input$arm_trt %in% ANL[[input$arm_var]],
+          paste0("Selected Treatment ", input$arm_var, ", ", input$arm_trt, ", is not in the data (filtered out?)")
         )
-      }
+      )
       validate(need(all(c(input$arm_trt, input$arm_ref) %in% unique(ANL[[input$arm_var]])), "Plot loading"))
 
-      teal.code::chunks_push(
-        id = "variables call",
-        expression = bquote({
-          id <- .(as.name(anl_name))[["USUBJID"]]
-          arm <- .(as.name(anl_name))[[.(input$arm_var)]]
-          arm_N <- table(ADSL[[.(input$arm_var)]]) # nolint
-          trt <- .(input$arm_trt)
-          ref <- .(input$arm_ref)
-          anl_labels <- formatters::var_labels(.(as.name(anl_name)), fill = FALSE)
-          flags <- .(as.name(anl_name)) %>%
+      q1 <- teal.code::eval_code(
+        teal.code::new_quosure(data),
+        name = "variables call",
+        code = as.expression(c(
+          bquote(anl_labels <- formatters::var_labels(.(as.name(dataname)), fill = FALSE)),
+          bquote(flags <- .(as.name(dataname)) %>%
```

Comment:
simplified the code to avoid making proxy-variables in the quosure.
Used two elemens expression to avoid `{` in the code